modules/aggregations/src/yamlRestTest/resources/rest-api-spec/test/aggregations/nested_with_scores.yml

```diff
@@ -0,0 +1,156 @@
+                my_buckets:
+                  composite:
+                    size: 20
+                    sources:
+                      -
+                        prod_name:
+                          terms:
+                            field: "lines.name"
+                  aggregations:
+                    catalog_price:
+                      scripted_metric:
+                        init_script:
+                          source: "state.transactions = []"
+                        map_script:
+                          source: "state.transactions.add(doc['lines.value'].value)"
+                        combine_script:
+                          source: "double sum = 0; for (t in state.transactions) { sum += t } return sum"
+                        reduce_script:
+                          source: "double sum = 0; for (a in states) { if (a != null) { sum += a } } return sum"
+---
+"Query, Composite, Scripted Metric (scores disabled)":
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          size: 0
+          query:
+            bool:
+              filter:
+                -
+                  range:
+                    "@timestamp":
+                      from: "2022-09-01"
+                      to: "2022-09-30"
+                      include_lower: true
+                      include_upper: true
+                      time_zone: "Europe/Paris"
+                      boost: 1
+              adjust_pure_negative: true
+              boost: 1
+          _source: true
+          track_total_hits: -1
+          aggregations:
+            data:
+              nested:
+                path: "lines"
+              aggregations:
+                my_buckets:
+                  composite:
+                    size: 20
+                    sources:
+                      -
+                        prod_name:
+                          terms:
+                            field: "lines.name"
+                  aggregations:
+                    catalog_price:
+                      scripted_metric:
+                        init_script:
+                          source: "state.transactions = []"
+                        map_script:
+                          source: "state.transactions.add(doc['lines.value'].value)"
+                        combine_script:
+                          source: "double sum = 0; for (t in state.transactions) { sum += t } return sum"
+                        reduce_script:
+                          source: "double sum = 0; for (a in states) { if (a != null) { sum += a } } return sum"
+                        needs_scores: false
+
+  - match: {aggregations.data.doc_count: 3}
+  - length: { aggregations.data.my_buckets.buckets: 2 }
+  - match: { aggregations.data.my_buckets.buckets.0.key.prod_name: "bar" }
+  - match: { aggregations.data.my_buckets.buckets.0.doc_count: 1}
+  - match: { aggregations.data.my_buckets.buckets.0.catalog_price.value: 2.0}
```

Comment:
I don't know why the scripted metric is losing resolution here.  It's troubling, and I haven't had time to dig into it.  It's also almost definitely unrelated, and probably indicates that I did something wrong in my script.